tests/pos/match-type-disjoint-22076.scala

```diff
@@ -0,0 +1,8 @@
+trait Foo[ CP <: NonEmptyTuple ]:
+  type EndNode = Tuple.Last [CP]
+
+def f( end : Foo[?] ) : end.EndNode =
```

Comment:
Consider adapting the coding style to what we actually use:
```suggestion
trait Foo[CP <: NonEmptyTuple]:
  type EndNode = Tuple.Last[CP]

def f(end: Foo[?]): end.EndNode =
```
The coding style in the OP is really weird.